scalafmt-cli/src/main/scala/org/scalafmt/cli/CliOptions.scala

```diff
@@ -92,13 +92,42 @@ case class CliOptions(
     gitOpsConstructor: AbsoluteFile => GitOps = x => new GitOpsImpl(x),
     noStdErr: Boolean = false
 ) {
+  private[this] val DefaultGit = false
+  private[this] val DefaultFatalWarnings = false
+  private[this] val DefaultIgnoreWarnings = false
+  private[this] val DefaultEncoding = Codec.UTF8
+
+  private[this] val tempConfigPath: Option[Path] = configStr.map { s =>
+    val file = Files.createTempFile(".scalafmt", ".conf")
+    Files.write(file, s.getBytes)
+    file
+  }
```

Comment:
Create a temporary file that contains configuration string specified by `--config-str`. This temporary file will be passed to `scalafmt-dynamic`.
See https://github.com/scalameta/scalafmt/pull/1367#issuecomment-464744077